src/plugins/felixhayashi/tiddlymap/js/widget/MapWidget.js

```diff
@@ -1931,8 +1931,8 @@ class MapWidget extends Widget {
 
     const viewLabel = this.view.getLabel();
 
-    // we do not used the cashed version since we need a new object!
-    const nodeData = this.view.getNodeData(node.id, true) || {};
+    // we copy the object since we intend to modify it.
+    const nodeData = Object.assign({}, this.view.getNodeData(node.id));
     // we need to delete the positions so they are not reset when a user
     // resets the style…
     delete nodeData.x;
```

Comment:
This is one change I'm not completely sure about. I'm not sure what it meant with "need a new object". It looks like the MapWidget wants to modify the nodeData, and that's why?

I've switched it to shallow cloning the nodeData so it can remove `nodeData.x` and `nodeData.y` without affecting the underlying cached object. Because there shouldn't be any case where the cached object itself is modified.

In fact, I was considering deep freezing all of ViewAbstraction's cached objects. Tiddlywiki does the same with tiddlers. They're never modified except by addTiddler, which refreshes the cache.

Reply:
> I'm not sure what it meant with "need a new object". It looks like the MapWidget wants to modify the nodeData, and that's why?
yeah, I want to persist the style but I delete the x,y coordinates before persisting. If I'd do that operation on the actual nodeData that is used in the view, the coordinates would be modified and lost. so I created a fresh copy via the `true` param in `getNodeData(node.id, true)` and do the operations on this object.

Your solution is fine for the x and y case, because the int primitive types are copied by value, not by reference. yet, if we'd ever needed to delete properties of nested objects in `nodeData`, the Object.assign would not suffice anymore and this would cause side effects by deleting nested props. Anyhow, I am fine with this change, please add a todo though `// note: a deep copy would be needed if nested properties were modified`.

also please use this syntax: `{ ...this.view.getNodeData(node.id) }`. thanks :)